power_grid_model_c/power_grid_model/include/power_grid_model/auxiliary/dataset.hpp

```diff
@@ -171,14 +171,34 @@ template <dataset_type_tag dataset_type_> class Dataset {
     template <class StructType>
     using DataStruct = std::conditional_t<is_data_mutable_v<dataset_type>, StructType, StructType const>;
 
-    // for columnar buffers, Data* data is empty and attributes is filled
+    // for columnar buffers, Data* data is empty and attributes.data is filled
     // for uniform buffers, indptr is empty
     struct Buffer {
         using Data = Dataset::Data;
 
         Data* data{nullptr};
         std::vector<AttributeBuffer<Data>> attributes{};
         std::span<Indptr> indptr{};
+        Idx find_attribute(std::string_view attr_name) const {
+            if (data == nullptr && std::ranges::all_of(attributes, [](auto const& x) { return x.data == nullptr; })) {
+                return invalid_index;
+            }
+
+            auto const found = std::ranges::find_if(
+                attributes, [attr_name](auto const& x) { return x.meta_attribute->name == attr_name; });
+
+            if (found == attributes.cend()) {
+                return invalid_index;
+            }
+            return std::distance(attributes.cbegin(), found);
+        }
+        template <typename T> const T* get_col_data_at_index(Idx index) const {
+            assert(data == nullptr);
+            if (data != nullptr) {
```

Comment:
Why have both?

Reply:
One only works in debug

Reply:
Only assert works in debug, but the runtime error gets thrown both time right?

Reply:
The first line could indeed be removed